app/src/main/java/org/mate/state/executables/ActionsScreenState.java

```diff
@@ -390,8 +390,10 @@ public boolean differentColor(IScreenState visitedState) {
 
                         found = true;
                         if (!wOther.getColor().equals(wThis.getColor())) {
```

Comment:
Use logical operator && to combine the nested if conditions.

Reply:
done